utils/spdx/src/main/kotlin/SpdxExpression.kt

```diff
@@ -261,7 +261,9 @@ class SpdxCompoundExpression(
             }
         }
 
-        return flattenedChildren.singleOrNull() ?: SpdxCompoundExpression(operator, flattenedChildren)
+        val flattened = flattenedChildren.singleOrNull() ?: SpdxCompoundExpression(operator, flattenedChildren)
+        val choices = flattened.validChoices()
```

Comment:
As you added explanatory comments in the previous commit I think one here would also be nice to explain why the choices are taken into account.